### DISCUSSION TITLE
Engines Search API parameters does not work

### DISCUSSION THREAD

**Discussion by smiralexan** [searx/searx#2770] [General]
As said [here](https://searx.github.io/searx/dev/search_api.html) there are some parameters for GET requests for specifying search engines. The problem is I don't understand how they works, or maybe it's a bug or something. Changing "engines" parameter doesn't affect to anything, same engines are used in the searx response. Changing "enabled_engines" and "disabled_engines" mostly leads to "search error". Even with my own instance when some search engine set to enabled in settings.yml, only specifing one this engine in "disabled_engines" leads to "search error". Can you explain me how to use these parameters properly?

**Comment by virtadpt** [comment]
Like this:

http://localhost:8888/?q=cattle%20mutilation%20maryland&engines=wp,bi,wd,ddg,wb,ws

**Reply by smiralexan** [reply]
I understood what's wrong. It you specify both 'categories' and 'engines', 'engines' parameter does not work. I think it's a bug.

Proof:
Only 'engines' - works.
![image](https://user-images.githubusercontent.com/52520363/115134545-5127a780-a03b-11eb-95ed-3676f7c41114.png)

With 'categories' - does not work.
![image](https://user-images.githubusercontent.com/52520363/115134573-89c78100-a03b-11eb-95bf-fd98f64da318.png)

**Reply by virtadpt** [reply]
Huh! I'm going to try it on my instance and see if that happens.

**Reply by virtadpt** [reply]
I just got it to work: http://localhost:8888/?q=cattle%20mutilation%20maryland&engines=wp,bi,wd,ddg,wb,ws&categories=images
